File: C/lt_01_estseq/lt01_estseq_25.c
#include <stdio.h>

int main() {
  int horaI, horaF, minI, minF;
  do {
    printf("Insira a Hora de inicio em (HH:MM): ");
    scanf("%d:%d", &horaI, &minI);
  } while ((horaI < 0) || (horaI >= 24) || (minI < 0) || (minI >= 60));

  do {
    printf("Insira a Hora do final em (HH:MM): ");
    scanf("%d:%d", &horaF, &minF);
  } while ((horaF < 0) || (horaF >= 24) || (minF < 0) || (minF >= 60));

  horaI *= 60;
  horaF *= 60;

  minI = minI + horaI;
  minF = minF + horaF;
  printf("\n%d -- %d\n", minI, minF);

  if (minF < minI) {
    minF = minF - minI + (60 * 24);
  } else {
    minF = minF - minI;
  }

  horaF = minF / 60;
  minF = minF % 60;

  printf("O tempo de jogo foi de %.2d:%.2d.\n", horaF, minF);
  return 0;
}

File: Python/lt_01_estseq/lt01_estseq_25.py
text=input("Insira a Hora do Inicio: ")
hora_inicio=int(text[:2])
minuto_inicio=int(text[3:])

text=input("Insira a Hora do Final: ")
hora_final=int(text[:2])
minuto_final=int(text[3:])

minuto_inicio = hora_inicio*60
minuto_final = hora_final*60

if minuto_final > minuto_inicio:
    minuto_final-=minuto_inicio
elif minuto_final==minuto_inicio:
    minuto_final-=minuto_inicio
elif minuto_final < minuto_inicio:
    minuto_final=minuto_final - minuto_inicio + 1440

hora_final=int(minuto_final/60)
minuto_final-=(hora_final*60)
print("O tempo de Jogo é",hora_final,":",minuto_final,".")

